src/algorithms/fardetectors/FarDetectorTrackerCluster.h

```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+// Copyright (C) 2023, Simon Gardner
+
+#pragma once
+
+#include <DDRec/CellIDPositionConverter.h>
+// Event Model related classes
+#include <edm4eic/RawTrackerHitCollection.h>
+#include <edm4hep/TrackerHitCollection.h>
+
+#include "FarDetectorTrackerClusterConfig.h"
+#include "algorithms/interfaces/WithPodConfig.h"
+
+namespace eicrecon {
+
+  class FarDetectorTrackerCluster : public WithPodConfig<FarDetectorTrackerClusterConfig>  {
+
+  public:
+
+    /** One time initialization **/
+    void init(const dd4hep::Detector* det,
+              std::shared_ptr<const dd4hep::rec::CellIDPositionConverter>,
+              std::shared_ptr<spdlog::logger>);
+
+    /** Event by event processing **/
+    std::unique_ptr<edm4hep::TrackerHitCollection> produce(const edm4eic::RawTrackerHitCollection &inputhits);
```

Comment:
Is it intentional that you skip the hit reconstruction step here? Normally we have the sequence `SimTrackerHit` -> `RawTrackerHit` -> `TrackerHit` -> `Cluster`. The hit reconstruction turns digitized hits back into something that has a position and energy deposition, not just an ADC channel under `charge`. This is relevant since the `charge` field includes a pedestal that will bias your clustering by over-weighting the lower energy outer hits in a cluster.

Reply:
There are several bits of the current early tracking chain and datamodel which I'm not very uncomfortable with, it feels like it's been built around getting the central silicon hits into the acts framework without much scope for generalisation. I'll try and summarize here but most points/questions don't belong in this PR.

1. A parameterization of charge sharing amongst pixels in silicon detectors (very technology dependent) needs to be added before digitization. Most of the current hits in the simulation will be singles but for e.g. AC-LGADs (I believe) this will be O(10+) more. [side note - is this being included in the DAQ rates?]. This factory is mostly a placeholder for when this can be introduced and different alternatives tested (including something through acts)
2.  The SiliconTrackerDigi algorithm doesn't allow for more than a single hit in a cell per event, this won't work with detectors with high occupancy over the few micro-second events, some integration time parameter could be added. Other configurable parameters for the digitization of each detector would be nice for studying but not necessary.
3.  The Clustering here is really an alternative to the TrackerHit step used in the rest of the trackers, in SiliconTrackerDigi there isn't currently any pedestal introduced and I would assume this would be subtracted and gain corrected by a calibration as part of a future digitization implementation. I don't see the benefit to converting back to positions and charge deposited for individual hits, clustering is done by cellid not position and the real TrackerHit energy and position can be a non-linear result of the clustering of neighbouring pixels.
4. The TrackerHitReconstruction algorithm doesn't currently feel like it's contributing anything, I feel like it it misusing the edm4eic::TrackerHit type as it contains the position and position error in two separate frames so if someone tried pass the collection somewhere else it isn't self consistent. Instead the edm4hep::TrackerHitPlane type looks like it keeps information on both local and global but is missing the nice timeError field.

I might be missing something else about how the data model is designed but it would be nice to keep some more relationships/associations along the way.

I'd be happy to be told I've interpreted something wrongly

TLDR; this is an alternative trackerHit step not skipping it. (Until there are improved digitization models)

Reply:
I see the charge sharing is included in the document, I should have checked first.